File: diophantine.js
const solequa = n => {
	const res = [];

	for (let i = 1; i < Math.sqrt(n); i++) {
		if (n % i !== 0) continue;
    
		const j = n / i;
		const y = (j - i) / 4;
		const x = i + 2 * y;
    
		if (
			x >= 0 && y >= 0 
      && (j === x + 2 * y) && (i == x - 2 * y) 
      && Number.isInteger(x) && Number.isInteger(y)
		)
			res.push([x, y]);
		
    
		
	}
  
	return res.sort((a, b) => b[0] - a[0]);
};

File: diophantine.py
import math


def sol_equa(n):
	res = []
	for i in range(1, int(math.sqrt(n)) + 1):
		if n % i == 0:
			j = n // i
			if (i + j) % 2 == 0 and (j - i) % 4 == 0:
				x = (i + j) // 2
				y = (j - i) // 4
				res.append([x, y])

	return res

# def sol_equa(n):
#   return [[(n/i + i)/2,(n/i - i)/4] for i in range(1, int(n ** 0.5) + 1)
#   if n % i == 0 and (n/i + i) % 2 ==0 and (n/i - i) % 4 ==0]
